doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+
+# Revision
+| Rev  |    Date    |       Author        | Change Description                                           |
+|:--:|:--------:|:-----------------:|:------------------------------------------------------------:|
+| 0.1  | | Rajesh Sankaran   | Initial version                              |
+| 0.2  | | Rajesh Pukhraj Jain | Added Warm boot requirement/design details |
+| 0.4  | | Tapash Das, Hasan Naqvi | Added L3 VXLAN details |
+| 0.5  | | Karthikeyan A | ARP and ND suppression |
+| 0.6  | | Kishore Kunal | Added Fdbsycnd details |
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+| **Term** | **Meaning**                               |
+| -------- | ----------------------------------------- |
+| BGP      | Border Gateway Protocol                   |
+| BUM      | Broadcast, Unknown unicast, and Multicast |
+| EVPN     | Ethernet Virtual Private Network          |
+| IMET     | Inclusive Multicast Ethernet Tag          |
+| IRB      | Integrated Routing and Bridging           |
+| L3GW     | Layer 3 Gateway                           |
+| NVO      | Network Virtualization Overlay            |
+| VNI      | VXLAN Network Identifier                  |
+| VRF      | Virtual Routing and Forwarding            |
+| VTEP     | VXLAN Tunnel End point                    |
+| VXLAN    | Virtual Extended LAN                      |
+
+# About this Manual
+
+This document provides general information about the EVPN VXLAN feature implementation based on RFC 7432 and 8365 in SONiC. 
+This feature is incremental to the SONiC.201904 release which is referred to as current implementation in this document.
+
+
+# 1 Introduction and Scope
+
+This document describes the Functionality and High level design of the EVPN VXLAN feature.
+
+The following are the benefits of an EVPN approach for VXLAN.
+
+ - Auto discovery of remote VTEPs, Auto configuration of tunnels and Auto extension of VLANs over VXLAN tunnels.
+ - Aids in VM mobility.
+ - Aids in detecting duplicate MACs in the network.
+ - Allows for dual homing support.
+ - Control plane MAC learning and ARP suppression leads to reduced flooding over an EVPN network.
+ - Coexists with the current EVPN L3 VXLAN approach. The same tunnel can be used to carry both L2 and L3 overlay traffic.
+ - Eases planning and configuration when supporting downstream assigned VNI for DCI usecases.
+
+In the current implementation, there is support for L3 forwarding over VXLAN and EVPN Type 5 route support based on VNET constructs. 
+
+This feature adds the following enhancements.
+
+- EVPN Type 2, 3 routes support.
+- VLAN extension over VXLAN tunnels.
+- Flooding of L2 BUM (Broadcast, Unknown Unicast, and Multicast) traffic in and out of VXLAN tunnel.
+- Forwarding of L2 unicast traffic in and out of VXLAN tunnel.
+- EVPN Type 5 route support based on VRF construct.
+- Routing of L3 (IPv4 and IPv6) traffic in and out of the VXLAN tunnel.
+- Overlay ECMP support.
+- EVPN ARP and ND suppression.
```

Comment:
Is distributed Anycast Gateway support, not a requirement?
What would be the default Gateway for a host in a given subnet?

Reply:
distributed Anycast Gateway will be supported.